account_invoice_supplierinfo_update_variant/wizard/wizard_update_invoice_supplierinfo_line.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, fields
+
+
+class WizardUpdateInvoiceSupplierinfoLine(models.TransientModel):
+    _inherit = 'wizard.update.invoice.supplierinfo.line'
+
+    to_variant = fields.Boolean(
+        string='Update to the variant',
+        help='- Check the option for an update on the product variant.\n'
+             '- Uncheck the option for an update on the product model.')
+
+    @api.multi
+    def _prepare_supplierinfo(self):
+        if self.to_variant:
+            return {
+                'product_id': self.product_id.id,
```

Comment:
https://github.com/OCA/account-invoicing/blob/d0987a68bfc9c86a59e811c6262327d71cbafcb1/account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo_line.py#L62-L69

The original method have the same values but changing just `product_id`:
```python
    @api.multi
    def _prepare_supplierinfo(self):
        self.ensure_one()
        return {
            'product_tmpl_id': self.product_id.product_tmpl_id.id,
            'name': self.wizard_id.invoice_id.supplier_partner_id.id,
            'min_qty': 0.0,
            'delay': 1,
        }
```

What about use result from `super` and change to custom values required in pro of easiest inheriting?

`res = supers(...); res.update({...}); res.pop()`

(I don't know maybe is not possible)